From 70198e66552b2dcb22465bb4a12058b7703a063d Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 26 Jan 2018 10:31:50 +0200
Subject: [PATCH 3301/5242] MGS-3632-1: drm: imx: dcss: adjust ratio when
 WR_SCL kicks in

commit  433b80d6b8ebd665f71c0e32b00975f965775cb7 from
https://source.codeaurora.org/external/imx/linux-imx.git

Using WRSCL for downscaling ratios between 3 and 5 can lead to more
DDR bandwidth beeing used (~400MB/s).

Hence, use WR_SCL only for downscaling ratios from 5 to 7.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-scaler.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-scaler.c b/drivers/gpu/imx/dcss/dcss-scaler.c
index 4905ad1..4f2ae49 100644
--- a/drivers/gpu/imx/dcss/dcss-scaler.c
+++ b/drivers/gpu/imx/dcss/dcss-scaler.c
@@ -322,8 +322,8 @@ struct dcss_scaler_ratios {
 
 static const struct dcss_scaler_ratios dcss_scaler_ratios[] = {
 	{max_downscale(3), max_upscale(5)},
-	{max_downscale(3), max_upscale(7)},
-	{max_downscale(3), max_upscale(7)},
+	{max_downscale(5), max_upscale(7)},
+	{max_downscale(5), max_upscale(7)},
 };
 
 static const struct dcss_scaler_ratios dcss_scaler_wrscl_ratios[] = {
-- 
1.7.9.5

